homeassistant/components/tado/config_flow.py

```diff
@@ -159,6 +160,56 @@ async def async_step_import(
             },
         )
 
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a reconfiguration flow initialized by the user."""
+        self.config_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reconfigure_confirm()
+
+    async def async_step_reconfigure_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a reconfiguration flow initialized by the user."""
+        errors: dict[str, str] = {}
+        assert self.config_entry
+
+        if user_input is not None:
+            user_input[CONF_USERNAME] = self.config_entry.data[CONF_USERNAME]
+            try:
+                await validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except PyTado.exceptions.TadoWrongCredentialsException:
+                errors["base"] = "invalid_auth"
+            except NoHomes:
+                errors["base"] = "no_homes"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+            if "base" not in errors:
+                return self.async_update_reload_and_abort(
+                    self.config_entry,
+                    data={**self.config_entry.data, **user_input},
```

Comment:
```suggestion
                    data=user_input,
```
those seem to have the same keys, so `user_input` would overwrite all of `self.config_entry.data` anyway

Reply:
Reverting this back, whereas this seem to mess up the test cases. Also, other reconfigure flows exactly implement in this way. My main aim was to update properly the new data.

Reply:
Oh this is strange. Why is the import flow adding `home_id` to data, but the standard config flow isn't? Imho this would render the `self._async_abort_entries_match` in the import flow useless.
But I guess this would be for a follow-up PR - or just to sit it out until July when the yaml import is deprecated anyway 🙃

Reply:
You require any further additions or have any questions, in order to complete this PR?